src/NUnitFramework/tests/Internal/ParamNumberConversionsTests.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections;
+
+namespace NUnit.Framework.Internal
+{
+    [TestFixture]
+    public class ParamNumberConversionTests
+    {
+        [TestCase(typeof(short))]
+        [TestCase(typeof(byte))]
+        [TestCase(typeof(sbyte))]
+        [TestCase(typeof(decimal))]
+        public void TestInts(Type targetType)
+        {
+            var data = new object[] { 1, 2, 3 };
+
+            IEnumerable result = ParamNumberConversions.ConvertDataToNumeric(data, targetType);
+
+            CollectionAssert.AllItemsAreInstancesOfType(result, targetType);
+        }
+
+        [Test]
+        public void TestStringToDecimal()
+        {
+            var data = new object[] { "0.1", "1.0", "-1.0" };
+            var expected = new decimal[] { 0.1M, 1.0M, -1.0M };
+
+            IEnumerable result = ParamNumberConversions.ConvertDataToNumeric(data, typeof(decimal));
+
+            Assert.That(result, Is.EquivalentTo(expected));
```

Comment:
Do these tests definitely fail in the correct way? I'm suspicious that `Assert.That(result, Is.EquivalentTo(data));` may also pass - due to NUnits non-strict type comparison in `NUnitEqualityComparer`. If so, it may be better to do something like `Assert.That(result, Is.EquivalentTo(expected).And.All.TypeOf<decimal>());`. Same applies to the DateTime test below.

Reply:
(I may be wrong here - I haven't tested this!)